vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditReferenceController.java

```diff
@@ -0,0 +1,65 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.asu.diging.vspace.core.model.IReference;
+import edu.asu.diging.vspace.core.model.impl.Reference;
+import edu.asu.diging.vspace.core.services.IReferenceManager;
+
+@Controller
+public class EditReferenceController {
+
+    @Autowired
+    private IReferenceManager referenceManager;
+
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/biblio/{biblioId}/reference/edit", method = RequestMethod.POST)
+    public ResponseEntity<String> editReference(@PathVariable("id") String slideId,
+            @PathVariable("moduleId") String moduleId, @PathVariable("biblioId") String biblioId, 
+            @RequestBody String ref) throws IOException {
+        
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode rootNode = mapper.readTree(ref);
+        
+        String title = rootNode.get("title").asText();
+        String author = rootNode.get("author").asText();
+        String year = rootNode.get("year").asText();
+        String journal = rootNode.get("journal").asText();
+        String url = rootNode.get("url").asText();
+        String volume = rootNode.get("volume").asText();
+        String issue = rootNode.get("issue").asText();
+        String pages = rootNode.get("pages").asText();
+        String editor = rootNode.get("editors").asText();
+        String type = rootNode.get("type").asText();
+        String note = rootNode.get("note").asText();
+        String id = rootNode.get("id").asText();
+        
+        IReference reference = referenceManager.getReference(id);
```

Comment:
throw 404 if not exist

Reply:
added HttpStatus.NOT_FOUND

Reply:
still not sure what you mean. do you mean "updated" not "outdated"?

Reply:
I have added HTTP.NOT_FOUND but this is not showing up here, if you do not see this in this file then i guess there is something wrong, that its not showing the change I made in this file, because I addressed this comment.

Reply:
<img width="661" alt="Screenshot 2024-09-25 at 15 32 40" src="https://github.com/user-attachments/assets/8f7d3105-9402-457d-8c51-60aa7e0fa194">
can you see HTTP.NOT_FOUND line

Reply:
Is this the unaddressed you commented about, last? or can you see the HTTPStatus.NOT_FOUND on your side, because it is not showing up on the review side for me but it is there in the branch's respective file.